dask/dataframe/tests/test_groupby.py

```diff
@@ -3190,3 +3191,88 @@ def test_groupby_slice_getitem(by, slice_key):
     # column projection after read_parquet etc
     assert hlg_layer(got.dask, "getitem")
     assert_eq(expect, got)
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        "min",
+        "max",
+        "sum",
+        "prod",
+        "first",
+        "last",
+        "idxmin",
+        "idxmax",
+    ],
+)
+@pytest.mark.parametrize(
+    "numeric_only",
+    [None, True, False],
+)
+def test_groupby_numeric_only_supported(func, numeric_only):
+    pdf = pd.DataFrame({"A": [1, 1, 2], "B": [3, 4, 3], "C": ["a", "b", "c"]})
+
+    ddf = dd.from_pandas(pdf, npartitions=3)
+
+    kwargs = {} if numeric_only is None else {"numeric_only": numeric_only}
+
+    expect_type_error = False
+    expect_future_warning = False
+    expected = None
+    try:
+        with warnings.catch_warnings(record=True) as w:
+            expected = getattr(pdf.groupby("A"), func)(**kwargs)
+            if len(w) > 0 and w[0].category == FutureWarning:
+                expect_future_warning = True
+    except FutureWarning:
+        expect_future_warning = True
+    except TypeError:
+        expect_type_error = True
+
+    # TODO: idxmax sometimes does not return the same results. See
+    # https://github.com/dask/dask/issues/9882.
+    # Until that's fixed, skip equality check.
+    check_equality = False if func == "idxmax" else True
+
+    if expect_type_error:
+        with pytest.raises(TypeError):
+            getattr(ddf.groupby("A"), func)(**kwargs)
+    elif expect_future_warning:
+        with pytest.warns(FutureWarning):
+            actual = getattr(ddf.groupby("A"), func)(**kwargs)
+            if expected is not None and check_equality:
+                assert_eq(expected, actual)
+    else:
+        actual = getattr(ddf.groupby("A"), func)(**kwargs)
+        if check_equality:
+            assert_eq(expected, actual)
+
+
+# TODO: this test will fail when we implement support for numeric_only=False for the following agg functions
+@pytest.mark.parametrize(
+    "func",
+    ["cumsum", "cumprod", "mean", "median", "var", "std", "corr", "cov"],
+)
+@pytest.mark.parametrize(
+    "numeric_only",
+    [
+        False,
+        pytest.param(
+            None,
+            marks=pytest.mark.skipif(
+                not PANDAS_GT_200, reason="requires pandas >= 2.0"
+            ),
+        ),
+    ],
+)
+def test_groupby_numeric_only_unsupported(func, numeric_only):
+    """These should throw an error if numeric_only is set to False"""
+    pdf = pd.DataFrame({"A": [1, 1, 2], "B": [3, 4, 3], "C": ["a", "b", "c"]})
+
+    ddf = dd.from_pandas(pdf, npartitions=3)
+
+    kwargs = {} if numeric_only is None else {"numeric_only": numeric_only}
+
+    with pytest.raises(NotImplementedError):
```

Comment:
Let's use `match="'numeric_only=False' is not implemented in Dask"` here to make sure the `NotImplementedError` being raise the the one we think it should be